[Akamai] - Migrated SIEM data stream from HTTPJSON to CEL input

ShourieG opened the pull request:
## Type of change
- Enhancement
- Breaking Change

## Proposed commit message
```
Migrated SIEM data stream from HTTPJSON to CEL input with 
with necessary reworks and working system tests. Updated 
minimum stack version to 8.18 to allow usage of required CEL functions. 
```
<!-- Mandatory
Explain here the changes you made on the PR.

Please explain:

- WHAT: patterns used, algorithms implemented, design architecture, message processing, etc.
- WHY:  the rationale/motivation for the changes

This text will be pasted into the squash dialog when the change is committed and will be
a long term historical record of the change to help future contributors understand the
change, please help them by making it clear and comprehensive, they may be you.

If the commit title is adequate to describe both of these things, The text here may be omitted
or replaced with "See title". The title of the PR will be used as the commit message title when
the merge is made and the "See title" marker will be removed if present.

The text here and the PR title will be subject to the PR review process.
-->

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).
- [ ] I have verified that any added dashboard complies with Kibana's [Dashboard good practices](https://docs.elastic.dev/ux-guidelines/data-viz/dashboard-good-practices) 

## Author's Checklist

<!-- Recommended
Add a checklist of things that are required to be reviewed in order to have the PR approved
-->
- [ ] 

## How to test this PR locally

<!-- Recommended
Explain here how this PR will be tested by the reviewer: commands, dependencies, steps, etc.
-->

## Related issues

- Closes https://github.com/elastic/integrations/issues/11338
- Relates https://github.com/elastic/integrations/issues/11338

## Screenshots

<!-- Optional
Add here screenshots presenting:
- Kibana UI forms presenting configuration options exposed by the integration
- dashboards with collected metrics or logs
-->

elasticmachine commented:
## :broken_heart: Build Failed
* [Buildkite Build](https://buildkite.com/elastic/integrations/builds/33054)
* Commit: ffd666ea298aa3a0c24618f4d5a2ea39dfed4dfa

### Failed CI Steps
* [Check integrations akamai](https://buildkite.com/elastic/integrations/builds/33054#019a0f7c-a7d4-4ed3-9d94-4ea11f8f3921)

### History
* :broken_heart: [Build #33009](https://buildkite.com/elastic/integrations/builds/33009) failed 175690c91adf884dfcf688b33eaad7fe7bda5e99

cc @ShourieG

<!--buildkite-pr-comment-integrations
{"builds":[{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/33054","number":33054,"commit":"ffd666ea298aa3a0c24618f4d5a2ea39dfed4dfa"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/33009","number":33009,"commit":"175690c91adf884dfcf688b33eaad7fe7bda5e99"}],"number":33054}
buildkite-pr-comment-->

efd6 commented on packages/akamai/changelog.yml:
```diff
@@ -1,4 +1,9 @@
 # newer versions go on top
+- version: "2.29.0"
+  changes:
+    - description: Migrated siem data stream from Httpjson to CEL.
```
```suggestion
    - description: Migrated siem data stream from HTTP JSON to CEL.
```

efd6 commented on packages/akamai/data_stream/siem/agent/stream/cel.yml.hbs:
```diff
@@ -0,0 +1,142 @@
+program: |
+  state.with(
+    (
+      state.?cursor.?recovery_mode.orValue(false) ? 
```
```suggestion
      state.?cursor.recovery_mode.orValue(false) ? 
```
similar throughout; after a `.?` operator, all the subsequent accesses are on optional types, so after the first the `?` is redundant.

efd6 commented on packages/akamai/data_stream/siem/agent/stream/cel.yml.hbs:
```diff
@@ -0,0 +1,142 @@
+                    string(u.Path) + "?" +
+                    string(u.RawQuery) + 
+                    "\t\t\t" + sig_base
+                  ).as(to_sign,
```
use `sprintf`

efd6 commented on packages/akamai/data_stream/siem/agent/stream/cel.yml.hbs:
run `celfmt` on this

efd6 commented on packages/akamai/data_stream/siem/agent/stream/cel.yml.hbs:
```diff
@@ -0,0 +1,142 @@
+            "cursor": {
+              ?"last_offset": lines.size() > 0 ? 
+                lines[lines.size() - 1].decode_json().as(lastEvent,
+                  has(lastEvent.offset) ? optional.of(lastEvent.offset) : optional.none()
```
```suggestion
                  lastEvent.?offset
```

elasticmachine commented:
## :broken_heart: Build Failed
* [Buildkite Build](https://buildkite.com/elastic/integrations/builds/33182)
* Commit: b4206f4510938addb383de744e4b479c76305c71

### Failed CI Steps
* [Check integrations akamai](https://buildkite.com/elastic/integrations/builds/33182#019a244a-16df-445a-99a2-e230c3f66afd)

### History
* :green_heart: [Build #33125](https://buildkite.com/elastic/integrations/builds/33125) succeeded d05fc95ec29c52657fde6fcc705e60557eb8fd14
* :broken_heart: [Build #33054](https://buildkite.com/elastic/integrations/builds/33054) failed ffd666ea298aa3a0c24618f4d5a2ea39dfed4dfa
* :broken_heart: [Build #33009](https://buildkite.com/elastic/integrations/builds/33009) failed 175690c91adf884dfcf688b33eaad7fe7bda5e99

cc @ShourieG

<!--buildkite-pr-comment-integrations
{"builds":[{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/33182","number":33182,"commit":"b4206f4510938addb383de744e4b479c76305c71"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/33125","number":33125,"commit":"d05fc95ec29c52657fde6fcc705e60557eb8fd14"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/33054","number":33054,"commit":"ffd666ea298aa3a0c24618f4d5a2ea39dfed4dfa"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/33009","number":33009,"commit":"175690c91adf884dfcf688b33eaad7fe7bda5e99"}],"number":33182}
buildkite-pr-comment-->

efd6 commented on packages/akamai/data_stream/siem/agent/stream/cel.yml.hbs:
```diff
@@ -0,0 +1,159 @@
+                    string(u.Scheme),
+                    string(u.Host),
+                    string(u.Path),
+                    string(u.RawQuery),
```
```suggestion
                    u.Scheme,
                    u.Host,
                    u.Path,
                    u.RawQuery,
```

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/integrations/builds/33429)
* Commit: 7993a5e7c9d3b034fd04f5299bcfda74a737c19c

### History
* :green_heart: [Build #33247](https://buildkite.com/elastic/integrations/builds/33247) succeeded d97ebf28cc06983f4bb44da2e6c7c0b6ddbd2352
* :green_heart: [Build #33188](https://buildkite.com/elastic/integrations/builds/33188) succeeded 2d32dc50142e7b1b33b8ee5bb78a3dfb5e50485c
* :broken_heart: [Build #33182](https://buildkite.com/elastic/integrations/builds/33182) failed b4206f4510938addb383de744e4b479c76305c71
* :green_heart: [Build #33125](https://buildkite.com/elastic/integrations/builds/33125) succeeded d05fc95ec29c52657fde6fcc705e60557eb8fd14
* :broken_heart: [Build #33054](https://buildkite.com/elastic/integrations/builds/33054) failed ffd666ea298aa3a0c24618f4d5a2ea39dfed4dfa
* :broken_heart: [Build #33009](https://buildkite.com/elastic/integrations/builds/33009) failed 175690c91adf884dfcf688b33eaad7fe7bda5e99

cc @ShourieG

<!--buildkite-pr-comment-integrations
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/33429","number":33429,"commit":"7993a5e7c9d3b034fd04f5299bcfda74a737c19c"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/33247","number":33247,"commit":"d97ebf28cc06983f4bb44da2e6c7c0b6ddbd2352"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/33188","number":33188,"commit":"2d32dc50142e7b1b33b8ee5bb78a3dfb5e50485c"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/33182","number":33182,"commit":"b4206f4510938addb383de744e4b479c76305c71"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/33125","number":33125,"commit":"d05fc95ec29c52657fde6fcc705e60557eb8fd14"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/33054","number":33054,"commit":"ffd666ea298aa3a0c24618f4d5a2ea39dfed4dfa"}],"number":33429}
buildkite-pr-comment-->

ShourieG commented:
@efd6, addressed all the comments. Let's approve this once we have successfully tested with a live api, there seems to be some outstanding issues still. Let me know if the current changes are ok or not.

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/integrations/builds/34080)
* Commit: a4f0257522a324e63a9a436f8353f4623bd56ca4

### History
* :broken_heart: [Build #33886](https://buildkite.com/elastic/integrations/builds/33886) failed 68001cafe42bd012b48570ccb9366aeecd6dfdea
* :green_heart: [Build #33659](https://buildkite.com/elastic/integrations/builds/33659) succeeded 67eb579de10b078b254bbfae5831539908e4eac3
* :green_heart: [Build #33501](https://buildkite.com/elastic/integrations/builds/33501) succeeded 78c91b61693c45b00c15bf0fb57f76213a03abde
* :green_heart: [Build #33429](https://buildkite.com/elastic/integrations/builds/33429) succeeded 7993a5e7c9d3b034fd04f5299bcfda74a737c19c
* :green_heart: [Build #33247](https://buildkite.com/elastic/integrations/builds/33247) succeeded d97ebf28cc06983f4bb44da2e6c7c0b6ddbd2352
* :green_heart: [Build #33188](https://buildkite.com/elastic/integrations/builds/33188) succeeded 2d32dc50142e7b1b33b8ee5bb78a3dfb5e50485c

cc @ShourieG

<!--buildkite-pr-comment-integrations
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/34080","number":34080,"commit":"a4f0257522a324e63a9a436f8353f4623bd56ca4"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/33886","number":33886,"commit":"68001cafe42bd012b48570ccb9366aeecd6dfdea"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/33659","number":33659,"commit":"67eb579de10b078b254bbfae5831539908e4eac3"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/33501","number":33501,"commit":"78c91b61693c45b00c15bf0fb57f76213a03abde"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/33429","number":33429,"commit":"7993a5e7c9d3b034fd04f5299bcfda74a737c19c"},{"buildStatus":{"state":"passed","success":true,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/integrations/builds/33247","number":33247,"commit":"d97ebf28cc06983f4bb44da2e6c7c0b6ddbd2352"}],"number":34080}
buildkite-pr-comment-->
